### PR TITLE
Automatic setup for mac/linux or linux subsystem users

### PR DESCRIPTION
**Pull request by tdymel**
These scripts are supposed to help newcomers into the project. 
The setup script sets up the server in such a way, that no build files can be accidentally added when committing changes.

Huge parts of the setup script are stolen from Ahoi's wiki entries. But it was a little cryptic, so I had to squeeze out some information of Ahoi, such that things can get setup properly here.

This script should simplify the entry for most users into the project.

### CODE REVIEW

File: Tools/Linux/run.sh [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# Making sure that paths are correct
+DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )";
+cd $DIR/Source;
```
Comment (ahoiahoi) [inline Tools/Linux/run.sh]: you should 
a) always use ${} as variable accessor
b) not use semicolon at the end
c) use something like 'cd ${DIR} || exit "something wrong"'

File: Tools/Linux/setup.sh [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+DNS="user=$DB_USER";
+
+if [ "$DB_PW" != "" ]; then
+        DNS="user=$DB_USER;password=$DB_PW";
+fi;
+
+# Making sure that paths are correct
+DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )/../../";
+cd $DIR;
+
```
Comment (ahoiahoi) [inline Tools/Linux/setup.sh]: see above
also, the script may not be invoked from the directory it resides in

File: Tools/Linux/setup.sh [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+# Cleaning
+rm -rf ${DIR}/Source/NexusForever.AuthServer/bin;
+rm -rf ${DIR}/Source/NexusForever.StsServer/bin;
+rm -rf ${DIR}/Source/NexusForever.WorldServer/bin;
+
```
Comment (ahoiahoi) [inline Tools/Linux/setup.sh]: never use "rm -rf" in a script without making sure the the variable is not empty

File: Tools/Linux/kill.sh [hunk 4]
```diff
@@ -0,0 +1,2 @@
+pkill -f NexusForever.StsServer
+pkill -f NexusForever.AuthServer
```
Comment (ahoiahoi) [inline Tools/Linux/kill.sh]: add linefeed, and what about the world server?

Comment (tdymel) [reply Tools/Linux/kill.sh]: The worldserver is not started as decoppled to the prompt, i. e. with &

File: Tools/Linux/setup.sh (hunk 2 above)
Comment (tdymel) [reply Tools/Linux/setup.sh]: At the end u can see a "/../../" appended to the directory. This makes sure that the source directory is correct. I added an empty check tho

**Comment by tdymel** [comment]
Usage:
Setup:
./nexus.sh -s game_path table_path db_user [db_pw]
Kill:
./nexus.sh -k
Run:
./nexus.sh -r

**Pull request merged**